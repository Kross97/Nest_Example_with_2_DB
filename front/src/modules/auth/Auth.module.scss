.authBlock {
  display: flex;
  flex-direction: column;
  border: 1px solid gray;
  background-color: #ab4165;
  row-gap: 12px;
  padding: 30px;
  position: fixed;
  top: 50%;
  left: 50%;
  transform: translateX(-50%) translateY(-50%);
}