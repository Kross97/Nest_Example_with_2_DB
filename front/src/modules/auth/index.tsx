import cn from './Auth.module.scss';
import React, { ChangeEvent,  useState } from 'react';
import { FetchAgent } from '../../FetchService';
import { useAllContext } from '../../provider/AllProvider';


export const AuthBlock = () => {
  const [data, setUser] = useState<Record<string, string>>({});

  const { setToken, setCurrentPortHandler } = useAllContext();

  const change = (e: ChangeEvent<HTMLInputElement>) => {
    const nameField = e.target.name;
    const value = e.target.value;
    setUser((prev) => ({ ...prev, [nameField]: value }));
  };

  const signIn = async () => {
     const tokenData = await FetchAgent.postRequest<{ token: string; currentPort: number}>({ url: 'auth/signIn', requestType: 'json', body: data });
     console.log('tokenData =>', tokenData)
     setToken(tokenData.token);
     setCurrentPortHandler(tokenData.currentPort);
  };

  return <div className={cn.authBlock}>
    <h3>Авторизация</h3>
    <span>Логин</span>
    <input onChange={change} type={'text'} name={'login'} value={data.login}/>
    <span>Пароль</span>
    <input onChange={change} type={'password'} name={'password'} value={data.password}/>
    <button onClick={signIn}>Войти</button>
  </div>;
};