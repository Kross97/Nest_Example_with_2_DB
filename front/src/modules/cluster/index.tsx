import React, { useEffect, useState } from "react";
import cn from "./Cluster.module.scss";
import { FetchAgent } from "../../FetchService";
import { IPortsData, useAllContext } from "../../provider/AllProvider";

export function ClusterBlock() {
  const [currentPorts, setPorts] = useState<number[]>([]);

  const { setCurrentPortHandler, currentPort } = useAllContext();

  const getCurrentPorts = () => {
    FetchAgent.getRequest({ url: "/childProcess_cluster/cluster_ports" })
      .then((results: IPortsData) => {
        setPorts(results.availablePorts);
      })
      .catch((err) => {
        console.log("Ошибка получения портов", Object.entries(err));
      });
  };

  useEffect(() => {
    getCurrentPorts();
  }, []);

  const enableClusterHandle = () =>
    FetchAgent.getRequest({ url: "/childProcess_cluster/cluster_enable" }).then(() => {
      getCurrentPorts();
    });

  const disableClusterHandler = () => {
    FetchAgent.getRequest({ url: "/childProcess_cluster/cluster_exit" }).then(() => {
      getCurrentPorts();
      setCurrentPortHandler();
    });
  };

  const setNewPortServer = () => {
    void setCurrentPortHandler();
  };

  return (
    <div className={cn.clusterBlock}>
      <h2>Работа с кластеризацией сервера (Cluster модуль node)</h2>
      <h3>{`Состояние кластеризации: ${currentPorts.length === 1 ? "Отключена" : "Включена"}`}</h3>
      <h3>{`Текущий порт: ${currentPort}`}</h3>
      <div>
        <span>Доступные порты:</span>
        <span>
          {currentPorts.map((currentPort) => (
            <React.Fragment key={currentPort}>
              <span style={{ cursor: "pointer", color: "blue" }} onClick={() => setCurrentPortHandler(currentPort)}>
                {currentPort}
              </span>
              <span>{` , `}</span>
            </React.Fragment>
          ))}
        </span>
      </div>
      <div>
        <button disabled={currentPorts.length === 1} onClick={setNewPortServer}>
          Выбрать новый порт сервера
        </button>
        <button disabled={currentPorts.length !== 1} onClick={enableClusterHandle}>
          Включить кластеризацию
        </button>
        <button disabled={currentPorts.length === 1} onClick={disableClusterHandler}>
          Отключить кластеризацию
        </button>
      </div>
    </div>
  );
}
