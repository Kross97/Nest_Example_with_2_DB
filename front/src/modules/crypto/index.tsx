import cn from './Crypto.module.scss';
import React, { useEffect, useState } from 'react';
import { FetchAgent } from '../../FetchService';


export const CryptoBlock = () => {
  const [search, setSearch] = useState('');
  const [cipherData, setCipherData] = useState('');
  const [deCipherData, setDeCipherData] = useState('');


  const cipherHandler = async () => {
   const res = await FetchAgent.postRequest({ url: 'crypto/cipher1', body: search})
   setCipherData(res);
  };

  const deCipherHandler = async () => {
    const res = await FetchAgent.postRequest({ url: 'crypto/decipher1', body: cipherData })
    setDeCipherData(res);
  };

  return  <div className={cn.cryptoBlock}>
    <h3>Модуль для работы с шифрованием (Crypto node.js)</h3>
    <h4>Шифрование и де-шифрование данных</h4>
    <label className={cn.column}>
      Данные для шифрования
      <input type={'text'} value={search} onChange={(e) => setSearch(e.target.value)} />
    </label>
    <div className={cn.column}>
      <span>Данные после шифрования</span>
      <span>{cipherData}</span>
    </div>
    <div className={cn.column}>
      <span>Данные после де-шифрования</span>
      <span>{deCipherData}</span>
    </div>
    <button onClick={cipherHandler}>шифровать</button>
    <button onClick={deCipherHandler}>де-шифровать</button>
  </div>
};