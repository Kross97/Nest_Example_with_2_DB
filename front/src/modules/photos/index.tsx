import cn from './Photos.module.scss';
import React, {ChangeEvent, SyntheticEvent, useEffect, useState} from 'react';
import {FetchAgent} from '../../FetchService';

interface IFile {
    id: number;
    name: string;
    mimeType: string;
}

export const PhotoBlock = () => {
    const [file, setFile] = useState<File | null | undefined>(null);
    const [files, setFiles] = useState<IFile[]>([]);
    const [search, setSearch] = useState('');

    useEffect(() => {
        FetchAgent.getRequest({url: '/media/all', body: { search }}).then((results) => {
            setFiles(results || []);
        });
    }, [search]);


    const change = (e: SyntheticEvent) => {
        const file = (e.target as HTMLInputElement).files?.[0];
        const formData = new FormData();
        if (file) {
            formData.set('file', file);
            FetchAgent.postRequest({url: '/media', body: formData});

        }
    };

    const changeMany = (e: SyntheticEvent) => {
        const files = (e.target as HTMLInputElement).files;
        const formData = new FormData();

        if (files) {
            for (let i = 0; i < files.length; i++) {
                formData.append('file', files[i]);
            }
            FetchAgent.postRequest({url: '/media/many', body: formData});
        }
    };

    const changeUserPhoto = (e: SyntheticEvent) => {
        const file = (e.target as HTMLInputElement).files?.[0];
        setFile(file);
    };
    const createUser = () => {
        if (file) {
            const formData = new FormData();
            formData.append('file', file);
            formData.set('user', JSON.stringify({
                name: {
                    first: 'front_photo_first_name',
                    last: 'front_photo_last_name',
                }, test: 'test ', car: {
                    model: 'vw',
                    year: 2001,
                },
            }));

            FetchAgent.postRequest({url: '/user/createWithFile', body: formData});

        }
    };
    const downloadFile = async (file: IFile) => {
        const arrayBuffer = await FetchAgent.getRequest({
            url: `/media/download/first/${file.id}`,
            responseType: 'arrayBuffer',
        });
        const blob = new Blob([arrayBuffer], {
            type: file.mimeType,
        });

        const a = document.createElement('a');
        a.href = URL.createObjectURL(blob);
        a.download = file.name;
        a.click();
    };

    const downloadFile2 = async (file: IFile) => {
        const arrayBuffer = await FetchAgent.getRequest({
            url: `/media/download/second/${file.id}`,
            responseType: 'arrayBuffer',
        });
        const blob = new Blob([arrayBuffer], {
            type: file.mimeType,
        });

        const a = document.createElement('a');
        a.href = URL.createObjectURL(blob);
        a.download = file.name;
        a.click();
    };

    const changeHandler = (e: ChangeEvent<HTMLInputElement>) => {
        setSearch(e.target.value);
    };


    return <div className={cn.photosBlock}>
        <h3>Модуль для работы с медиа-материалами (Сохранение и получение медиа из БД)</h3>
        <h3>Фотографии</h3>
        <div className={cn.column}>
            <span>Закачка медиа</span>
            <input onChange={change} type={'file'}/>
        </div>
        <div className={cn.column}>
            <span>Закачка медиа (множественное)</span>
            <input onChange={changeMany} multiple={true} type={'file'}/>
        </div>
        <div className={cn.column} style={{alignItems: 'flex-start'}}>
            <span>Установка пользователя вместе с фотографией</span>
            <input onChange={changeUserPhoto} type={'file'}/>
            {file && <button onClick={createUser}>Создать пользователя</button>}
        </div>
        <label style={{display: 'flex', flexDirection: 'column', rowGap: '10px'}}>
            <b>Поиск</b>
            <input type={'text'} onChange={changeHandler}/>
        </label>
        <div className={cn.listFiles}>
            {files.map((f: IFile) => {
                return <div
                    style={{
                        display: 'flex',
                        flexDirection: 'column',
                        whiteSpace: 'wrap',
                        wordBreak: 'break-all',
                        rowGap: '12px'
                    }}>
                    <b>{f.name}</b>
                    <button style={{alignSelf: 'flex-start'}} onClick={() => downloadFile(f)}>загрузка_способ_1</button>
                    <button style={{alignSelf: 'flex-start'}} onClick={() => downloadFile2(f)}>загрузка_способ_2
                    </button>
                </div>;
            })}
        </div>
    </div>;
};