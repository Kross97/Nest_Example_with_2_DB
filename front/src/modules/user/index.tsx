import cn from './User.module.scss';
import React, { ChangeEvent, useEffect, useState } from 'react';
import { FetchAgent } from '../../FetchService';
import { IUser, IUSerRequest } from './types';
import { ModalUser } from './ModalUser';


export const UserBlock = () => {
  const [userEdit, setUserEdit] = useState<IUSerRequest | null>(null);
  const [users, setUsers] = useState<IUser[]>([]);
  const [data, setUser] = useState<Record<string, string>>({});

  const showEditHandler = (user: IUSerRequest | null) => setUserEdit(user);

  useEffect(() => {
    FetchAgent.getRequest({ url: '/user/all' }).then((results) => {
      setUsers(results);
    });
  }, []);

  const change = (e: ChangeEvent<HTMLInputElement>) => {
    const nameField = e.target.name;
    const value = e.target.value;
    setUser((prev) => ({ ...prev, [nameField]: value }));
  };

  const submitUser = async () => {
    await FetchAgent.postRequest({ url: '/user/create', body: data, requestType: 'json' });
  };

  const refreshUsers = async () => {
    await FetchAgent.getRequest({ url: '/user/refresh' });
  };

  return <>
    <ModalUser user={userEdit} showEditHandler={showEditHandler} />
    <div className={cn.userBlock}>
      <div className={cn.formUser}>
        <span>Name</span>
        <input onChange={change} type={'text'} name={'nameFirst'} value={data.nameFirst} />
        <span>LastName</span>
        <input onChange={change} type={'text'} name={'nameLast'} value={data.nameLast}/>
        <span>Логин</span>
        <input onChange={change} type={'text'} name={'login'} value={data.login}/>
        <span>Пароль</span>
        <input onChange={change} type={'password'} name={'password'} value={data.password}/>
        <button onClick={submitUser}>Создать пользователя</button>
      </div>
      <button onClick={refreshUsers}>рефреш_логинов</button>
      <div className={cn.listUsers}>
        {users.map((userList) => <div>
          <span>{`${userList.name.first} ${userList.name.last}`}</span>
          <button onClick={() => showEditHandler({
            nameFirst: userList.name.first,
            nameLast: userList.name.last,
            login: userList.login,
            password: userList.password,
            id: userList.id,
            role: userList.role
          })}>изменить
          </button>
        </div>)}
      </div>
    </div>
  </>;
};