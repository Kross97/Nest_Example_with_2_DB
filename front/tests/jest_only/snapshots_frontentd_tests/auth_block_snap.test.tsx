import React from 'react';
import { AuthBlock } from '../../../src/modules/auth';
import renderer from 'react-test-renderer';

describe('Тестирование снапшотами блока Auth', () => {
    it('Снапшот компонента модалки ', () => {
        const tree = renderer.create(<AuthBlock></AuthBlock>).toJSON();
        expect(tree).toMatchSnapshot();
        expect(tree).toMatchInlineSnapshot(`
            <div
              className="authBlock"
            >
              <h3
                data-testid="auth_header"
              >
                Авторизация (после CI\\CD)
              </h3>
              <label
                style={
                  {
                    "display": "flex",
                    "flexDirection": "column",
                  }
                }
              >
                Логин
                <input
                  name="login"
                  onChange={[Function]}
                  type="text"
                  value=""
                />
              </label>
              <span>
                Пароль
              </span>
              <input
                name="password"
                onChange={[Function]}
                type="password"
                value=""
              />
              <button
                onClick={[Function]}
                role="button"
              >
                Войти
              </button>
            </div>
        `);
    });
});
