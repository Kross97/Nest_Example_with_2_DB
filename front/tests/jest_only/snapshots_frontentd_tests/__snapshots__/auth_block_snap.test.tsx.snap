// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Тестирование снапшотами блока Auth Снапшот компонента модалки  1`] = `
<div
  className="authBlock"
>
  <h3
    data-testid="auth_header"
  >
    Авторизация (после CI\\CD)
  </h3>
  <label
    style={
      {
        "display": "flex",
        "flexDirection": "column",
      }
    }
  >
    Логин
    <input
      name="login"
      onChange={[Function]}
      type="text"
      value=""
    />
  </label>
  <span>
    Пароль
  </span>
  <input
    name="password"
    onChange={[Function]}
    type="password"
    value=""
  />
  <button
    onClick={[Function]}
    role="button"
  >
    Войти
  </button>
</div>
`;
